scripts/xtensa-build-zephyr.py

```diff
@@ -25,6 +25,7 @@
 elif platform.system() == "Linux":
 	xtensa_tools_version_postfix = "-linux"
 else:
+	xtensa_tools_version_postfix = "-unsupportedOS"
```

Comment:
Is this needed?

Reply:
It's needed otherwise the script crashes on Mac. With it cloning works just fine, this entire PR was written and tested on a Mac. I never tried it myself but Zephyr can be compiled on a Mac. Even on Linux the XTENSA toolchain is not required, I use it only when I really need it because of the licensing hassle.

Reply:
> It's needed otherwise the script crashes on Mac. 

Did you look at the commit messages?

Reply:
Yeah overlooked it my bad.
Ok it can be as "for-now" solution because it is just not very elegant.
I mean the MAC user will just get a error indicating that a toolchain is not found under a given path (not that it is not supported). And in this given path will be just this -unsupportedOS syntax. But we can change it at later stages when all this CI hassle will be complete :-)

> Even on Linux the XTENSA toolchain is not required, I use it only when I really need it because of the licensing hassle.

Well, some *our-Intel* products do require XTENSA toolchain and ZephyrSDK is explicitly *not-supported*.
As far as I know those xtensa compilers (the one in ZephyrSDK and the one in the commercial distribution) are very different. The fact that you can always build with ZephyrSDK does not mean that it will meet our performance goals.

Reply:
> The fact that you can always build with ZephyrSDK does not mean that it will meet our performance goals.

Good point and good reminder but: 1) the Zephyr SDK is still useful for testing. Tests that are not performance sensitive are already passing! More compilers / analyzers can find more warnings and more bugs 2) SOF and Zephyr are not just an Intel project; who knows what others want to support now or in the future 3) We hope to make some progress with other toolchains. 4) Drive-by contributors may not have an XCC license. 5) Github Actions do not have an  XCC license for now.

While only one toolchain will ever be officially supported at a time, the more toolchains and static analyzers can compile and run the code the merrier - as long as they don't require complex `#ifdef` hacks or workarounds or too much work. XCC toolchains are based on old versions of either gcc or clang; so far XCC toolchains were the ones requiring ugly `#ifdef`hacks to disable standard C features and a lot of work, just ask @dcpleung who did the work to get Zephyr to compile with XCC...